sql/core/src/test/scala/org/apache/spark/sql/test/DataFrameReaderWriterSuite.scala

```diff
@@ -635,4 +638,13 @@ class DataFrameReaderWriterSuite extends QueryTest with SharedSQLContext with Be
       checkAnswer(spark.table("t"), Row(1, "a") :: Row(2, "b") :: Nil)
     }
   }
+
+  test("use saveAsTable to append to a data source table implementing CreatableRelationProvider") {
+    withTable("t") {
+      val provider = "org.apache.spark.sql.test.DefaultSource"
```

Comment:
I am thinking that we probably should create a test data source that explicitly mentions its relation is not an InsertableRelation.